### PR TITLE
Fix minimum import time in partition synopses

### PR DESCRIPTION
**Pull request by dominiklohmann**
Turns out that `min(epoch, t)` is always `epoch`. This fixes that by assigning `time::max()` for the minimum import time.

### :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Not really user-facing (yet), so no changelog entry needed. The fix is quite easy to understand.

### CODE REVIEW

**Review by lava — APPROVED** [review]

**Comment by dominiklohmann** [comment]
@lava feel free to merge by yourself after approving of the additional unit tests.

**Comment by dominiklohmann** [comment]
[sc-29848]

File: libvast/src/system/meta_index.cpp [hunk 1]
```diff
@@ -276,34 +276,18 @@ std::vector<uuid> meta_index_state::lookup_impl(const expression& expr) const {
...
-              }
-              if (add)
+              VAST_ASSERT(part_syn.min_import_time <= part_syn.max_import_time);
+              auto ts = time_synopsis{
+                part_syn.min_import_time,
+                part_syn.max_import_time,
+              };
+              auto add = ts.lookup(x.op, caf::get<vast::time>(d));
+              if (!add || *add) {
+                VAST_WARN("yes: part {} w/ range [{}, {}]", part_id,
```
Comment (lava) [inline libvast/src/system/meta_index.cpp:286]: That looks like left-over debug code? (also, is this functionally the same as the previous version, or did it include a bug that was fixed by using the time synopsis?)

File: libvast/test/system/counter.cpp [hunk 2]
```diff
@@ -166,4 +168,76 @@ TEST(count IP point query with partition - local stores) {
...
+  auto indexdir = directory / "index2";
+  auto index
+    = self->spawn(system::index, system::accountant_actor{}, fs, archive,
+                  indexdir, "segment-store", defaults::import::table_slice_size,
+                  100, 3, 1, indexdir, 0.01);
+  // Fill the INDEX with 400 rows from the Zeek conn log.
+  auto slices = take(zeek_conn_log_full, 4);
+  for (auto& slice : slices) {
+    slice = slice.unshare();
+    slice.import_time(time::clock::now());
```
Comment (lava) [inline libvast/test/system/counter.cpp:182]: Do we have to set a valid offset here?

File: libvast/vast/detail/partition_common.hpp [hunk 3]
```diff
@@ -50,8 +51,26 @@ fetch_indexer(const PartitionState& state, const meta_extractor& ex,
         row_ids |= ids;
     }
   } else if (ex.kind == meta_extractor::age) {
-    for (const auto& [_, ids] : state.type_ids())
-      row_ids |= ids;
+    // For the passive partition we can trust the eval from the meta index, but
+    // we cannot rely on that for the active partition. We create a time
```
Comment (lava) [inline libvast/vast/detail/partition_common.hpp:55]: That was a bit confusing to understand, I'd suggest something like "For a passive partition, this already went through a time synopsis in the meta index, but for the active partition we create an ad-hoc time synopsis here to do the lookup."

File: libvast/test/system/meta_index.cpp [hunk 4]
```diff
@@ -268,6 +285,35 @@ TEST(attribute extractor - type) {
   CHECK_EQUAL(lookup("#type !~ /x/"), ids);
 }
 
+TEST(attribute extractor - age) {
```
Comment (lava) [inline libvast/test/system/meta_index.cpp:288]: Can you add a comment that half the test data was set to import age 1975 and the other half to 2015 in the test setup?

**Review by lava — APPROVED** [review]
Looks good module the unnecessary WARN() log message; the unit tests seem to cover the basic usage.

File: libvast/test/system/counter.cpp (hunk 2 above)
Comment (dominiklohmann) [reply libvast/test/system/counter.cpp:182]: Nope, `slice.unshare()` already takes care of that.

File: libvast/src/system/meta_index.cpp (hunk 1 above)
Comment (dominiklohmann) [reply libvast/src/system/meta_index.cpp]: > also, is this functionally the same as the previous version, or did it include a bug that was fixed by using the time synopsis?

Same behavior, just less code to maintain. I didn't want to duplicate it again, considering how much trouble we had wrapping our head around the seemingly simple comparison yesterday.

File: libvast/test/system/counter.cpp (hunk 2 above)
Comment (lava) [reply libvast/test/system/counter.cpp:182]: Only if the slice had a valid offset to begin with, but it doesn't seem to be relevant for this test anyways.

**Pull request merged**